patches/tModLoader/Terraria/ModLoader/PlayerLayerHooks.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+
+namespace Terraria.ModLoader
+{
+	public static class PlayerLayerHooks
```

Comment:
wait so, `PlayerDrawLayer`s register themselves and add themselves to the layer list, unless they're the special subclass which isn't autoloaded. This is confusing. I think this design here is good, but `LegacyPlayerDrawLayer` is out of line...